### PR TITLE
Fixes log filter to actually work

### PR DESCRIPTION
**Pull request by eikes**
The filter regexp didn't match the actual headers. Now there is a test which verifies that...

### CODE REVIEW

File: lib/ioki/http_adapter.rb [hunk 1]
```diff
@@ -10,20 +10,20 @@ def self.get(config)
       ::Faraday.new(config.api_base_url, headers: headers(config)) do |f|
         f.adapter :net_http
 
+        f.request :authorization, 'Bearer', -> { config.api_token }
+        f.request :json # encode req bodies as JSON
```
Comment (rmehner) [inline lib/ioki/http_adapter.rb:14]: Why is this back on top again? Doesn't this revert the changes of #44?

Comment (eikes) [reply lib/ioki/http_adapter.rb:14]: These two are the middleware for the outgoing `request`. The order of the middleware for the incoming `response` is what I changed in #44. The thing is, that the logger inspects both `resquest` and `response` even if it is only called on the response (don't ask me why) and I noticed, that the authorization header would not be logged if it is configured in this setup after the logger. I was not able to test yet how this order affects the logging of the outgoing request bodies, such as a POST request body.

I can take a look after the weekend though.

**Comment by eikes** [comment]
@rmehner I checked and the request body for POST / PATCH requests was not parsed before logging. Thanks for pointing that out. It did work regardless, even for non-ascii characters already though, because the request was already using UTF-8. Anyhow, I changed it and the request formatting of the json body in the log now matches the response.

**Review by A5308Y — APPROVED** [review]
